go/coordinator/internal/metastore/coordinator/memory_catalog.go

```diff
@@ -14,94 +14,262 @@ import (
+	log.Info("collection created", zap.Any("collection", collection))
+	collections[collection.ID] = collection
 	return collection, nil
 }
 
-func (mc *MemoryCatalog) GetCollections(ctx context.Context, collectionID types.UniqueID, collectionName *string, collectionTopic *string) ([]*model.Collection, error) {
-	collections := make([]*model.Collection, 0, len(mc.Collections))
-	for _, collection := range mc.Collections {
+func (mc *MemoryCatalog) GetCollections(ctx context.Context, collectionID types.UniqueID, collectionName *string, collectionTopic *string, tenantID string, databaseName string) ([]*model.Collection, error) {
+	if _, ok := mc.tenantDatabaseCollections[tenantID]; !ok {
+		log.Error("tenant not found", zap.String("tenant", tenantID))
+		return nil, common.ErrTenantNotFound
+	}
+	if _, ok := mc.tenantDatabaseCollections[tenantID][databaseName]; !ok {
+		log.Error("database not found", zap.String("database", databaseName))
+		return nil, common.ErrDatabaseNotFound
+	}
+	collections := make([]*model.Collection, 0, len(mc.tenantDatabaseCollections[tenantID][databaseName]))
+	for _, collection := range mc.tenantDatabaseCollections[tenantID][databaseName] {
 		if model.FilterCollection(collection, collectionID, collectionName, collectionTopic) {
 			collections = append(collections, collection)
 		}
 	}
-	log.Debug("collections", zap.Any("collections", collections))
 	return collections, nil
 }
 
-func (mc *MemoryCatalog) DeleteCollection(ctx context.Context, collectionID types.UniqueID) error {
-	if _, ok := mc.Collections[collectionID]; !ok {
+func (mc *MemoryCatalog) DeleteCollection(ctx context.Context, deleteCollection *model.DeleteCollection) error {
+	tenantID := deleteCollection.TenantID
+	databaseName := deleteCollection.DatabaseName
+	collectionID := deleteCollection.ID
+	if _, ok := mc.tenantDatabaseCollections[tenantID]; !ok {
+		log.Error("tenant not found", zap.String("tenant", tenantID))
+		return common.ErrTenantNotFound
+	}
+	if _, ok := mc.tenantDatabaseCollections[tenantID][databaseName]; !ok {
+		log.Error("database not found", zap.String("database", databaseName))
+		return common.ErrDatabaseNotFound
+	}
+	collections := mc.tenantDatabaseCollections[tenantID][databaseName]
+	if _, ok := collections[collectionID]; !ok {
+		log.Error("collection not found", zap.String("collection", collectionID.String()))
 		return common.ErrCollectionDeleteNonExistingCollection
 	}
-	delete(mc.Collections, collectionID)
+	delete(collections, collectionID)
+	log.Info("collection deleted", zap.String("collection", collectionID.String()))
 	return nil
 }
 
-func (mc *MemoryCatalog) UpdateCollection(ctx context.Context, coll *model.UpdateCollection, ts types.Timestamp) (*model.Collection, error) {
-	oldCollection := mc.Collections[coll.ID]
-	topic := coll.Topic
+func (mc *MemoryCatalog) UpdateCollection(ctx context.Context, updateCollection *model.UpdateCollection, ts types.Timestamp) (*model.Collection, error) {
```

Comment:
This is not why, collections are unique by id in order to give some unique identifier for lookup and convenience. Before tenants/database, this was critically unique and the api path'ed on collection id.  This had been the case since v0 of chroma. It had nothing to do with the tenant/database concept.

Update and Delete operate on the id because thats what they already supported. They could be extended to take the full qualifier but I don't think this is needed rn. 

Lets break it down:

CreateCollection -> We **have** to specify ID + tenant/database/name
DeleteCollection -> Before tenant and database this only took ID, not name. Since IDs were globally unique. It still only takes ID and I added tenant/db as future proofing. 
GetCollection -> Must take, and always took id or tenant/database/name
UpdateCollection -> Only works on ID, as that how it worked before. IDs had to be globally unique, I should have likely added tenant/db to the proto but this is strictly additive and not consumed.

Upstream calls operate on the ID, so it was built this way and then migrated.